### PR TITLE
[Refactor]: create RawCurp and Context by builder pattern

### PR DESCRIPTION
**Pull request by themanforfree**
Please briefly answer these questions:
 create RawCurp and Context by builder pattern
* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

**Review by Phoenix500526 — DISMISSED** [review]
LGTM

**Review by Phoenix500526 — APPROVED** [review]
LGTM

File: curp-external-api/src/role_change.rs [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
+#![allow(clippy::integer_arithmetic, clippy::indexing_slicing)] //  introduced by mock!
+
+use mockall::mock;
 
 /// Callback when the leadership changes
-#[allow(clippy::indexing_slicing)]
-#[allow(clippy::integer_arithmetic)]
-#[automock]
-pub trait RoleChange: Send + Sync + 'static {
+pub trait RoleChange: Clone + Send + Sync + 'static {
```
[the remark this answers is not in the document]
Comment (themanforfree) [reply curp-external-api/src/role_change.rs:6]: The builder macro requires all generic types implement Clone, even if they are wrapped in Arc

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Review by iGxnon — APPROVED** [review]

**Pull request merged**